src/arch/metal.rs

```diff
@@ -621,27 +622,61 @@ _start:
     add t0, t0, t3       // The actual start of our stack
     add t1, t0, t1       // And the end of our stack
 
+    ld t4, __bss_start
+    ld t5, __bss_stop
+
     // Then we fill the stack with a known memory pattern
     li t2, 0x0BADBED0
+
 loop:
-    bgeu t0, t1, done // Exit when reaching the end address
+    bgeu t0, t1, zero_loop // Exit when reaching the end address
     sw t2, 0(t0)      // Write the pattern
     addi t0, t0, 4    // increment the cursor
     j loop
-done:
 
+zero_loop:
+    bgeu t4, t5, verify
+    sb x0, 0(t4)
+    addi t4, t4, 1
+    j zero_loop
+
+verify:
+    ld t4, __bss_start
+    ld t5, __bss_stop
+    j verify_loop
+
+verify_loop:
+    bgeu t4, t5, done
+    lb t6, 0(t4)  
+    beqz t6, continue_verification
```

Comment:
It seems there is a logic issue here:
If `t6` is zero, then we go to continue verification. But if `t6` is _not_ zero the branch is not taken and the PC moves to the next instruction, which is also `continue_verification`.
So in other words the verification doesn't verify anything  😅

Reply:
yes, there should be something like bnez t6, halt_execution
it probably doesn't matter already though